### DISCUSSION TITLE
BundlerMinifier vs CDN

### DISCUSSION THREAD

**Discussion by kappawng** [marcominerva/TinyCMS#2] [General]
Ciao @marcominerva,
perchè preferisci usare BundlerMinifier al posto delle CDN?
Solo una questione di _tutorial_?

Grazie

**Comment by marcominerva** [comment]
Sicuramente mostrare **BundleMinifier** era interessante per l'argomento del Live, ma nello specifico ho preferito utilizzare questa libreria proprio per non dover mettere in piedi un discorso di CDN per i file CSS e JS specifici del sito.

**Reply by kappawng** [reply]
Comunque mi riferivo al fatto di linkare direttamente per esempio Bootstrap da jsdelivr.net, tipo
`<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.min.js"></script>`

Non so se avevo posto la discussione nel modo giusto :-D

Grazie comunque per la risposta, aspettiamo la seconda parte del live ;-)

**Reply by marcominerva** [reply]
Ah OK, quando hai parlato di **BundleMinifier** pensavo ti riferissi appunto ai file del sito (che sono quelli per cui lo uso), non alle librerie esterne ;-)

Anche per le librerie esterne c'è un discorso di "tutorial", ma in questo caso ho voluto fare così perché la soluzione che stiamo sviluppando vorrei che fosse "self-contained" (passami l'espressione). Ovviamente, a dirla tutta, dovremmo anche considerare lo scenario in cui la CDN è down (ipotesi abbastanza rara, ma tecnicamente possibile), ma è anche vero che in questo caso avremmo potuto adottare una soluzione basata su CDN con un fallback su file locale in caso di non disponibilità, soluzione che però almeno per il momento ho preferito non menzionare, per concentrarmi maggiormente sull'applicazione in sé.

Ultimo, ma non meno importante, nel progetto abbiamo previsto di usare anche **Font Awesome**, la cui documentazione ufficiale suggerisce il self-hosting delle icone. Volendo quindi seguire le linee guida, mi piaceva poco l'idea di usare una CDN per **Bootstrap** e **jQuery** e un self-hosting per Font Awesome, quindi ho preferito gestire tutto in locale, naturalmente come hai visto utilizzando la CDN per gestire download e update delle librerie con un semplice click.

**Reply by kappawng** [reply]
OK, chiarissimo ;-)
